File: backend/src/main/java/com/poeflipfinder/backend/gateway/SnapshotQueryGateway.java
package com.poeflipfinder.backend.gateway;

import com.poeflipfinder.backend.entity.ExchangeMarketSnapshot;
import java.util.List;

/**
 * Read side of the generation-tag hot-swap (docs/SCHEMA.md § Ingestion state
 * and market data) -- deliberately separate from SnapshotRepositoryGateway,
 * which is write-only and owned by the ingestion use case (Interface
 * Segregation: different callers, different lifecycles).
 */
public interface SnapshotQueryGateway {

    /**
     * All snapshots in the currently-active generation for one league. Empty
     * if the league is unrecognized or no generation has ever gone live yet.
     */
    List<ExchangeMarketSnapshot> findActiveSnapshots(String leagueExternalId);
}

File: app/lib/poe_flip_finder/snapshot_query_gateway.ex
defmodule PoeFlipFinder.SnapshotQueryGateway do
  @moduledoc """
  Read side of the generation-tag hot-swap (docs/SCHEMA.md § Ingestion
  state and market data) -- deliberately separate from
  `PoeFlipFinder.SnapshotRepositoryGateway`, which is write-only and owned
  by the ingestion use case (Interface Segregation: different callers,
  different lifecycles).
  """

  alias PoeFlipFinder.ExchangeMarketSnapshot

  @doc """
  All snapshots in the currently-active generation for one league. Empty
  if the league is unrecognized or no generation has ever gone live yet.
  """
  @callback find_active_snapshots(league_external_id :: String.t()) :: [
              ExchangeMarketSnapshot.t()
            ]
end
